pkg/cluster/spec.go

```diff
@@ -116,16 +113,72 @@ func NewSpec(config *Config, bundles *v1alpha1.Bundles, eksdRelease *eksdv1alpha
 	return s, nil
 }
 
-func (s *Spec) KubeDistroImages() []v1alpha1.Image {
-	images := []v1alpha1.Image{}
-	for _, component := range s.eksdRelease.Status.Components {
-		for _, asset := range component.Assets {
-			if asset.Image != nil {
-				images = append(images, v1alpha1.Image{URI: asset.Image.URI})
-			}
+func getAllVersionsBundles(cluster *eksav1alpha1.Cluster, bundles *v1alpha1.Bundles, eksdReleases []eksdv1alpha1.Release) (map[eksav1alpha1.KubernetesVersion]*VersionsBundle, error) {
+	if len(eksdReleases) < 1 {
+		return nil, fmt.Errorf("no eksd releases were found")
+	}
+
+	m := make(map[eksav1alpha1.KubernetesVersion]*VersionsBundle, len(eksdReleases))
+
+	for _, eksd := range eksdReleases {
+		channel := strings.Replace(eksd.Spec.Channel, "-", ".", 1)
+		version := eksav1alpha1.KubernetesVersion(channel)
+
+		if _, ok := m[version]; ok {
+			continue
 		}
+
+		versionBundle, err := getVersionBundles(version, bundles, &eksd)
+		if err != nil {
+			return nil, err
+		}
+
+		m[version] = versionBundle
+	}
+
+	return m, nil
+}
+
+func getVersionBundles(version eksav1alpha1.KubernetesVersion, b *v1alpha1.Bundles, eksdRelease *eksdv1alpha1.Release) (*VersionsBundle, error) {
+	v, err := GetVersionsBundle(version, b)
+	if err != nil {
+		return nil, err
 	}
-	return images
+
+	kd, err := buildKubeDistro(eksdRelease)
+	if err != nil {
+		return nil, err
+	}
+
+	vb := &VersionsBundle{
+		VersionsBundle: v,
+		KubeDistro:     kd,
+	}
+
+	return vb, nil
+}
+
+// VersionsBundles returns a VersionsBundle if one exists for the provided kubernetes version and nil otherwise.
+func (s *Spec) VersionsBundles(version eksav1alpha1.KubernetesVersion) *VersionsBundle {
```

Comment:
ok maybe we need to synç offline, we keep coming back to this name. Sorry I have created any confusion. I think this should be
```suggestion
// VersionsBundle returns a VersionsBundle if one exists for the provided kubernetes version and nil otherwise.
func (s *Spec) VersionsBundle(version eksav1alpha1.KubernetesVersion) *VersionsBundle {
```
Because it returns one single versions bundle